### ISSUE TITLE
No devices discovered on Mac

### ISSUE THREAD

**Issue by epheterson** [emmcc/node-switchmate#7]
I got all of the packages installed nicely, but each time I run discovery, it returns no results. I've also tried running as sudo and it's the same. Any ideas?

~/.nvm/versions/node/v7.6.0/lib/node_modules/node-switchmate/bin/discover.switchmate

**Comment by psolin** [comment]
Same here - it just times out.  Xcode is installed along with all of the dependencies (followed the directions).  Sierra 10.12.6 2017 MacBook Pro.

**Comment by epheterson** [comment]
Dang at least it’s not just me. A friend got it working fine on his Raspberry Pi, so I suspect the issue is Mac specific. Please comment back if you figure it out! I’ve given up...

**Comment by psolin** [comment]
A dedicated Raspberry Pi is probably the way to go with Homebridge anyway, if you don't want to tie up a computer in use or on-the-go frequently.  I think that the problem might have something to do with the bluetooth scanning portion of the code.  In any case, it seems that a Wink 2 Hub might do the trick to get these devices on a home network, and then Siri integration may be possible after that (though I haven't tried).  Very unfortunate.

**Comment by epheterson** [comment]
Yeah I've heard the Wink plugin works well. For me, it's already working on my iMac and I leave it running all day anyway, so I'll likely just never see it work. Thinking of giving it to my friend who has it working on his Pi!

**Comment by armaanbindra** [comment]
The issue guys is not the mac, It used to work before the firmware update on my mac and raspberry pi, linux, and everywhere, but after the firmware update they completely changed the bluetooth keys, and connection settings, rendering this plugin useless for anybody who has the original versions of switchmate, so until we can somehow rediscover what the new settings are, this wont be fixed :( . This would all be so simple, if switchmate just provided a REST API, plugin interface or IFTT Recipe for their switches.

**Comment by epheterson** [comment]
But since they’re Bluetooth any IFTTT or Web API functionality would require a bridge, so don’t hold your breath...

**Comment by armaanbindra** [comment]
Well yeah, thats why when this worked, my raspberry pi was the bridge :P , I think they are planning on building a new one with either wifi inbuilt or a their own hub. They do work with Wink hubs I think. Yeah, with the old firmware I used a simple node server on the raspberry pi that was internet connected, letting me control the switches, via my own messenger, bot, alexa, and google home all simultaneously and smoothly, I even was about to get it to work with homekit, now that apple has relaxed and opened up that system a bit. But the darn firmware update :( , it basically forced me to do it, upon opening the app, and killed all the different integrations with it in one big swoop.

**Comment by epheterson** [comment]
I wonder if they realized people were using it and might put it back if requested. Has anyone asked?

**Comment by psolin** [comment]
Just saw this thread again.  The company claims that a software update where it will be compatible with the Wink Hub 2 again will be coming soon.  A previous firmware update definitely broke its compatibility with other systems.
